tswow-scripts/wotlk/std/CreatureOutfits/CreatureOutfits.ts

```diff
@@ -159,4 +165,109 @@ export class CreatureOutfit extends MainEntity<creature_template_outfitsRow> {
+        if(this.Wrists.get()) extraInfo.NPCItemDisplay.setIndex(7, std.Items.load(this.Wrists.get()).DisplayInfo.get())
+        if(this.Hands.get()) extraInfo.NPCItemDisplay.setIndex(8, std.Items.load(this.Hands.get()).DisplayInfo.get())
+        if(this.Tabard.get()) extraInfo.NPCItemDisplay.setIndex(9, std.Items.load(this.Tabard.get()).DisplayInfo.get())
+        if(this.Back.get()) extraInfo.NPCItemDisplay.setIndex(10, std.Items.load(this.Back.get()).DisplayInfo.get())
+
+        switch(this.Race.get()) {
+            case RaceIDs.HUMAN:
+                this.Gender.get() == 0 ? modelID = 49 : modelID = 50
+                break
+            case RaceIDs.DWARF:
+                boundingRadius =  0.347
+                this.Gender.get() == 0 ? modelID = 53 : modelID = 54
+                break
+            case RaceIDs.GNOME:
+                boundingRadius =  0.3519
+                combatReach = 1.725
+                this.Gender.get() == 0 ? modelID = 182 : modelID = 183
+                break
+            case RaceIDs.NIGHTELF:
+                boundingRadius =  0.389
+                this.Gender.get() == 0 ? modelID = 55 : modelID = 56
+                break
+            case RaceIDs.DRAENEI:
+                this.Gender.get() == 0 ? modelID = 2248 : modelID = 2250
+                break
+            case RaceIDs.ORC:
+                boundingRadius =  0.372
+                this.Gender.get() == 0 ? modelID = 51 : modelID = 52
+                break
+            case RaceIDs.TROLL:
+                this.Gender.get() == 0 ? modelID = 185: modelID = 186
+                break
+            case RaceIDs.UNDEAD:
+                boundingRadius =  0.383
+                this.Gender.get() == 0 ? modelID = 57 : modelID = 58
+                break
+            case RaceIDs.TAUREN:
+                boundingRadius =  0.9747
+                combatReach = 4.05
+                this.Gender.get() == 0 ? modelID = 59 : modelID = 60
+                break
+            case RaceIDs.BLOODELF:
+                boundingRadius = 0.383
+                this.Gender.get() == 0 ? modelID = 2208 : modelID = 2209
+                break
+        }
+
+        let displayInfo = DBC.CreatureDisplayInfo.add(Ids.CreatureDisplayInfo.dynamicId())
+            .ModelID.set(modelID)
+            .SoundID.set(0)
+            .ExtendedDisplayInfoID.set(extraInfo.ID.get())
+            .CreatureModelAlpha.set(255)
+            .CreatureModelScale.set(1)
+            .NPCSoundID.set(this.NPCSounds.get())
+
+        SQL.creature_model_info.add(displayInfo.ID.get())
+            .Gender.set(this.Gender.get())
+            .CombatReach.set(combatReach)
+            .BoundingRadius.set(boundingRadius)
+
+        return displayInfo
+    }
+
+    /**
+     * Adds a baked texture to the model
+     * @param texture file name of the .blp texture in Textures/BakedNpcTextures
+     * @returns
+     */
+    addTexture(
```

Comment:
Can this be named better, something like addBakedTexture, something to  better signify that this is for a specific purpose?